File: bwl.Network.ClientServer/CmdRemoting/CmdlineUi.vb
Imports System.Windows.Forms

Public Class CmdlineUi
    Private _client As CmdlineClient

    Public Sub New()

        ' Этот вызов является обязательным для конструктора.
        InitializeComponent()

        ' Добавить код инициализации после вызова InitializeComponent().

    End Sub
    Public Sub New(client As CmdlineClient)

        ' Этот вызов является обязательным для конструктора.
        InitializeComponent()
        _client = client
        ' Добавить код инициализации после вызова InitializeComponent().
        AddHandler client.OutputReceived, AddressOf BuffersRecievedHandler
    End Sub

    Private Sub BuffersRecievedHandler(standartOutput As String)
        Try
            Me.Invoke(Sub()
                          If cbFilter.Checked And tbFilter.Text > "" Then
                              Dim lines = standartOutput.Split({vbCr, vbLf}, StringSplitOptions.RemoveEmptyEntries)
                              For Each line In lines
                                  If line.ToLower.Contains(tbFilter.Text.ToLower) Then
                                      TextBox1.AppendText(line + vbCrLf)
                                  End If
                              Next
                          Else
                              TextBox1.AppendText(standartOutput)
                          End If
                      End Sub)
        Catch ex As Exception
        End Try
    End Sub

    Private Sub CmdlineUi_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub

    Private Sub cbInput_KeyDown(sender As Object, e As KeyEventArgs) Handles cbInput.KeyDown
        If e.KeyCode = Keys.Enter Then
            Dim line = cbInput.Text
            _client.SendStandartInput(line)
            cbInput.Text = ""
            e.SuppressKeyPress = True
        End If
    End Sub

    Private Sub timerUpdate_Tick(sender As Object, e As EventArgs) Handles timerUpdate.Tick
        Try
            _client.RequestUpdate()
        Catch ex As Exception

        End Try
        cbAlive.Checked = _client.ServerAlive
        cbHasExited.Checked = _client.HasExited
        cbHasStarted.Checked = _client.HasStarted
        cbResponding.Checked = _client.Responding
        '   Me.Text = "RemoteCmd " + _client.WindowTitle
    End Sub

    Private Sub CmdlineUi_FormClosing(sender As Object, e As FormClosingEventArgs) Handles Me.FormClosing
        RemoveHandler _client.OutputReceived, AddressOf BuffersRecievedHandler
    End Sub

    Private Sub bClear_Click(sender As Object, e As EventArgs) Handles bClear.Click
        TextBox1.Text = ""
    End Sub
End Class

File: ClientServer/ClientServer/Bwl.Network.ClientServer.Windows/CmdRemoting/CmdlineUi.cs
using System.Runtime.Versioning;
using Bwl.Framework;

namespace Bwl.Network.ClientServer.Windows
{

    [SupportedOSPlatform("windows")]
    public partial class CmdlineUi : Form, IUIWindow
    {
        private CmdlineClient _client;

        string IUIWindow.Text { get => this.Text; set => this.Text = value; }

        event EventHandler IUIWindow.Load
        {
            add { this.Load += value; }
            remove { this.Load -= value; }
        }

        private Dictionary<EventHandler, FormClosedEventHandler> _formClosedEventHandlerMap = new Dictionary<EventHandler, FormClosedEventHandler>();

        event EventHandler IUIWindow.FormClosed
        {
            add
            {
                FormClosedEventHandler handler = (s, e) => value(s, e);
                _formClosedEventHandlerMap.Add(value, handler);
                this.FormClosed += handler;
            }
            remove
            {
                if (_formClosedEventHandlerMap.TryGetValue(value, out var handler))
                {
                    this.FormClosed -= handler;
                    _formClosedEventHandlerMap.Remove(value);
                }
            }
        }

        public CmdlineUi()
        {
            InitializeComponent();
        }

        public CmdlineUi(CmdlineClient client)
        {
            InitializeComponent();
            _client = client;
            client.OutputReceived += this.BuffersRecievedHandler;
        }

        void IUIWindow.ShowDialog(object invokeForm)
        {
            base.ShowDialog((Form)invokeForm);
        }

        void IUIWindow.Show()
        {
            base.Show();
        }

        void IUIWindow.Close()
        {
            base.Close();
        }

        void IUIWindow.Invoke(Action action)
        {
            this.Invoke(action);
        }

        private void BuffersRecievedHandler(string standartOutput)
        {
            try
            {
                Invoke(() =>
                {
                    if (cbFilter.Checked && !string.IsNullOrEmpty(tbFilter.Text))
                    {
                        string[] lines = standartOutput.Split(new[] { "\r", "\n" }, StringSplitOptions.RemoveEmptyEntries);
                        foreach (var line in lines)
                        {
                            if (line.ToLower().Contains(tbFilter.Text.ToLower()))
                            {
                                TextBox1.AppendText($"{line}\r\n");
                            }
                        }
                    }
                    else
                    {
                        TextBox1.AppendText(standartOutput);
                    }
                });
            }
            catch (Exception ex)
            {
                // log exception if needed
            }
        }

        private void CmdlineUi_Load(object sender, EventArgs e)
        {

        }

        private void cbInput_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                string line = cbInput.Text;
                _client.SendStandartInput(line);
                cbInput.Text = "";
                e.SuppressKeyPress = true;
            }
        }

        private void timerUpdate_Tick(object sender, EventArgs e)
        {
            try
            {
                _client.RequestUpdate();
            }
            catch (Exception ex)
            {
                // log exception if needed
            }
            cbAlive.Checked = _client.ServerAlive;
            cbHasExited.Checked = _client.HasExited;
            cbHasStarted.Checked = _client.HasStarted;
            cbResponding.Checked = _client.Responding;
        }

        private void CmdlineUi_FormClosing(object sender, FormClosingEventArgs e)
        {
            _client.OutputReceived -= this.BuffersRecievedHandler;
        }

        private void bClear_Click(object sender, EventArgs e)
        {
            TextBox1.Text = "";
        }
    }
}